pkg/action/git-credential.go

```diff
@@ -0,0 +1,192 @@
+			return nil, err
+		}
+		key = strings.TrimSuffix(key, "=")
+		val, err := rd.ReadString('\n')
+		if err == io.EOF {
+			err = io.ErrUnexpectedEOF
+		}
+		if err != nil {
+			return nil, err
+		}
+		val = strings.TrimSuffix(val, "\n")
+		switch key {
+		case "protocol":
+			c.Protocol = val
+		case "host":
+			c.Host = val
+		case "path":
+			c.Path = val
+		case "username":
+			c.Username = val
+		case "password":
+			c.Password = val
+		}
+	}
+}
+
+// GitCredentialBefore is executed before another git-credential command
+func (s *Action) GitCredentialBefore(ctx context.Context, c *cli.Context) error {
+	if !ctxutil.IsStdin(ctx) {
+		return ExitError(ctx, ExitUsage, nil, "missing stdin from git")
+	}
+	return s.Initialized(ctx, c)
+}
+
+// GitCredentialGet returns a credential to git
+func (s *Action) GitCredentialGet(ctx context.Context, c *cli.Context) error {
+	ctx = sub.WithAutoSync(ctx, false)
+	cred, err := parseGitCredentials(os.Stdin)
+	if err != nil {
+		return ExitError(ctx, ExitUnsupported, err, "Error: %v while parsing git-credential", err)
+	}
+	// try git/host/username... If username is empty, simply try git/host
+	path := "git/" + fsutil.CleanFilename(cred.Host) + "/" + fsutil.CleanFilename(cred.Username)
+	if !s.Store.Exists(ctx, path) {
+		// if the looked up path is a directory with only one entry (e.g. one user per host), take the subentry instead
+		if s.Store.IsDir(ctx, path) {
+			tree, err := s.Store.Tree(ctx)
+			if err != nil {
+				return ExitError(ctx, ExitDecrypt, err, "Error: %v while listing the storage", err)
+			}
+			sub, err := tree.FindFolder(path)
+			if err != nil {
+				return ExitError(ctx, ExitDecrypt, err, "Error: %v while listing the storage", err)
+			}
+			entries := sub.List(0)
+			if len(entries) == 1 {
+				path = "git/" + entries[0]
+			} else {
+				fmt.Fprintln(os.Stderr, "gopass error: too many entries")
+			}
+		} else {
+			return nil
+		}
+	}
+	secret, err := s.Store.Get(ctx, path)
+	if err != nil {
+		return ExitError(ctx, ExitDecrypt, err, "")
+	}
+	cred.Password = secret.Password()
+	username, err := secret.Value("login")
+	if err != nil {
+		// leave the username as is otherwise
+		cred.Username = username
+	}
+
+	_, err = cred.WriteTo(out.Stdout)
+	if err != nil {
+		return ExitError(ctx, ExitIO, err, "Could not write to stdout")
+	}
+	return nil
+}
+
+// GitCredentialStore stores a credential got from git
+func (s *Action) GitCredentialStore(ctx context.Context, c *cli.Context) error {
+	ctx = sub.WithAutoSync(ctx, false)
+	cred, err := parseGitCredentials(os.Stdin)
+	if err != nil {
+		return ExitError(ctx, ExitUnsupported, err, "Error: %v while parsing git-credential", err)
+	}
+	path := "git/" + fsutil.CleanFilename(cred.Host) + "/" + fsutil.CleanFilename(cred.Username)
+	secret := secret.New(cred.Password, "")
+	if cred.Username != "" {
+		_ = secret.SetValue("login", cred.Username)
+	}
+	// TODO: Skip this step if the same secret is already in the store. How to implement this?
```

Comment:
How can comparison to existing keys be done?

Reply:
No, I'm looking for a way to check if the secret that would be stored equals the secret already stored by encrypting the current secret and comparing it to the stored secret (no need to decrypt the stored secret here)

Reply:
This is not supported. I don't think this edge case would be enough to extend the (already pretty big) store interface. If you want to compare it the old value will need to be decrypted.